### DISCUSSION TITLE
[코딩컨벤션] 주로 사용하는 어노테이션의 순서를 정하는 것은 어떨까요?

### DISCUSSION THREAD

**Discussion by Starlight258** [Attica-org/athens-backend#29] [General]
클래스 위에 여러 어노테이션이 붙는 경우가 많고, 주로 사용하는 어노테이션만 쓰다보니 어노테이션의 순서를 통일하면 좀 더 가독성이 좋을 것 같아요.
예를 들어 엔티티의 경우 getter,setter는 위로 올리고, 생성자는 아래쪽으로 두는 식으로? (어떤 위치던 통일을 하면 좋을 것 같아요)

```java
@Getter
@ToString(callSuper = true)
@Table(indexes = {
        @Index(columnList = "title"),
        @Index(columnList = "createdAt"),
        @Index(columnList = "createdBy")
})
@Entity
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Agora extends BaseTimeEntity {
```

그런데 어노테이션의 수가 많다보니 하나하나 정하는게 어려울 것 같기도 하네요 🥲
여러분의 생각이 궁금해요~! @big-cir @yybmion

**Comment by yybmion** [comment]
중복해서 쓰는 어노테이션이 많아서 정해놓으면 좋을 것 같기는 하지만 세세하게 정하면 정해야할 어노테이션이 많아서 고민이 되네요.

**Reply by Starlight258** [reply]
동감합니다,, 좋은 방법 없을까요?

**Reply by big-cir** [reply]
공통적으로 정의해주는 필수 어노테이션들만 선별해서 순서를 정해주고, 그 이외에는 하위에서 자유롭게 사용해도 괜찮을 것 같습니다.!
